nibabel/nicom/dicomwrappers.py

```diff
@@ -471,13 +494,28 @@ def image_shape(self):
         frame_indices = np.array(
             [frame.FrameContentSequence[0].DimensionIndexValues
              for frame in self.frames])
-        n_dim = frame_indices.shape[1] + 1
-        # Check there is only one multiframe stack index
-        if np.any(np.diff(frame_indices[:, 0])):
-            raise WrapperError("File contains more than one StackID. Cannot "
-                               "handle multi-stack files")
+        # Check that there is only one multiframe stack index
+        stack_ids = set(frame.FrameContentSequence[0].StackID
+                        for frame in self.frames)
+        if len(stack_ids) > 1:
+            raise WrapperError("File contains more than one StackID. "
+                               "Cannot handle multi-stack files")
+        # Determine if one of the Dimension indices refers to the stack id
+        dim_seq = [(dim.DimensionIndexPointer, dim.FunctionalGroupPointer)
+                   for dim in self.get('DimensionIndexSequence')]
+        # the pointer pair (StackID Dicom-tag, FrameContentSequence Dicom-tag)
+        # that indicates that the corresponding axis in the DimensionIndexValues
+        # attribute refers to the StackID
+        stack_id_dim_pointer = ((0x20, 0x9056), (0x20, 0x9111))
```

Comment:
I will update the references to point to the current standard.  I have been working off of the 2015c version of the standard that did not include the supplement. 

I agree that using _tag_for_name_ is preferable. I will rewrite like how you suggested.  This brings up another thought that occurred to me: do you think that it would be beneficial to have a method/attribute that returned/stored the DICOM attribute name that each axis of the data corresponds to?  A user could use the "DimensionIndexSequence" to determine these on their own, but doing it in such a manner could be error prone for them since we are removing the "StackID" axis.

I believe you are correct that the "FrameContentSequence" tag is not needed.  It does appear that it would be an error in the DICOM (C.7.6.17.1 Dimension Indices) if the "FunctionalGroupPointer" was something other than the "FrameContentSequence" tag of "StackID".  However, it does not appear that the DICOM standard guarantees that each axis will have a "FunctionalGroupPointer" attribute when one is not needed for that axis (i.e. the dimension attribute is not nested).  If this was the case for an axis, then how I have it coded up now will not work.  So i am going rewrite the code to just rely on the "DimensionIndexPointer".